ccl/multiregionccl: TestMultiRegionTenantRegions failed

ccl/multiregionccl.TestMultiRegionTenantRegions [failed](https://tanzanite.cluster.engflow.com/invocations/default/11cd5bc0-d521-4a19-a0a6-183bc6238f94?testReportRun=13&testReportShard=4&testReportAttempt=1#targets-Ly9wa2cvY2NsL211bHRpcmVnaW9uY2NsOm11bHRpcmVnaW9uY2NsX3Rlc3Q=) on master @ [4ee917141b193bd2db451258cd040dc3b423951e](https://github.com/cockroachdb/cockroach/commits/4ee917141b193bd2db451258cd040dc3b423951e):


```
=== RUN   TestMultiRegionTenantRegions
I231213 07:06:13.393106 1 (gostd) rand.go:250  [-] 1  random seed: -2014089149424895374
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestMultiRegionTenantRegions3430372297
    test_log_scope.go:81: use -show-logs to present logs inline
    sql_runner.go:87: error executing query="ALTER DATABASE system DROP REGION \"us-east2\"" args=[]: pq: count-leases: enum value "us-east2" is not yet public
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestMultiRegionTenantRegions3430372297
--- FAIL: TestMultiRegionTenantRegions (5.71s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=13</code>
 - <code>shard=4</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiRegionTenantRegions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34571

--- a/pkg/sql/logictest/testdata/logic_test/alter_type
+++ b/pkg/sql/logictest/testdata/logic_test/alter_type
@@ -214,7 +214,7 @@ SELECT enum_last('c'::build)
 ----
 f
 
-statement error pq: enum value \"g\" is not yet public
+statement error  pq: cannot use enum value \"g\": enum value is not yet public
 INSERT INTO new_enum_values VALUES ('g')
 
 statement ok

--- a/pkg/sql/logictest/testdata/logic_test/enums
+++ b/pkg/sql/logictest/testdata/logic_test/enums
@@ -1733,7 +1733,7 @@ statement ok
 ALTER TYPE greeting ADD VALUE 'salud' AFTER 'hello'
 
 # The insert should fail but with awareness that 'salud' is an enum type.
-statement error pq: enum value "salud" is not yet public
+statement error pq: cannot use enum value \"salud\": enum value is not yet public
 INSERT INTO tab2 VALUES ('salud')
 
 statement ok

--- a/pkg/sql/regionliveness/BUILD.bazel
+++ b/pkg/sql/regionliveness/BUILD.bazel
@@ -18,6 +18,7 @@ go_library(
         "//pkg/sql/sem/tree",
         "//pkg/sql/sessiondata",
         "//pkg/sql/sqlliveness/slinstance",
+        "//pkg/sql/types",
         "//pkg/util/timeutil",
         "@com_github_cockroachdb_errors//:errors",
     ],

--- a/pkg/sql/regionliveness/prober.go
+++ b/pkg/sql/regionliveness/prober.go
@@ -27,6 +27,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
 	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/slinstance"
+	"github.com/cockroachdb/cockroach/pkg/sql/types"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/errors"
 )
@@ -218,5 +219,6 @@ func IsQueryTimeoutErr(err error) bool {
 // IsMissingRegionEnumErr determines if a query hit an error because of a missing
 // because of the region enum.
 func IsMissingRegionEnumErr(err error) bool {
-	return pgerror.GetPGCode(err) == pgcode.InvalidTextRepresentation
+	return pgerror.GetPGCode(err) == pgcode.InvalidTextRepresentation ||
+		errors.Is(err, types.EnumValueNotYetPublicError)
 }

--- a/pkg/sql/types/types.go
+++ b/pkg/sql/types/types.go
@@ -2705,6 +2705,9 @@ func (t *T) EnumGetIdxOfPhysical(phys []byte) (int, error) {
 	return 0, err
 }
 
+// EnumValueNotYetPublicError enum value is not public yet.
+var EnumValueNotYetPublicError = errors.New("enum value is not yet public")
+
 // EnumGetIdxOfLogical returns the index within the TypeMeta's slice of
 // enum logical representations that matches the input string.
 func (t *T) EnumGetIdxOfLogical(logical string) (int, error) {
@@ -2717,7 +2720,7 @@ func (t *T) EnumGetIdxOfLogical(logical string) (int, error) {
 			// written until all nodes in the cluster are able to decode the
 			// physical representation.
 			if t.TypeMeta.EnumData.IsMemberReadOnly[i] {
-				return 0, errors.Newf("enum value %q is not yet public", logical)
+				return 0, errors.WithMessagef(EnumValueNotYetPublicError, "cannot use enum value %q", logical)
 			}
 			return i, nil
 		}

